packages/material-ui/src/utils/useEnhancedEffect.js

```diff
@@ -0,0 +1,11 @@
+import * as React from 'react';
+
+const useEnhancedEffect =
+  typeof window !== 'undefined' && process.env.NODE_ENV !== 'test'
```

Comment:
The env branch is a change in behavior. This makes test incredibly hard to reason about. Even worse: Components using this hook are now not tested how they would actually work.